### PR TITLE
Remove libc from dev deps

### PR DESCRIPTION
**Pull request by notgull**
https://github.com/smol-rs/polling/pull/139#issuecomment-1705642150

### CODE REVIEW

**Comment by fogti** [comment]
I'd prefer just re-exporting `rustix::process::Signal` itself...
```patch
diff --git a/Cargo.toml b/Cargo.toml
index 1d4e100..1a74b6b 100644
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -48,6 +48,3 @@ features = [
 [dev-dependencies]
 easy-parallel = "3.1.0"
 fastrand = "2.0.0"
-
-[target.'cfg(any(unix, target_os = "fuchsia", target_os = "vxworks"))'.dev_dependencies]
-libc = "0.2"
diff --git a/examples/wait-signal.rs b/examples/wait-signal.rs
index 4a645af..9410526 100644
--- a/examples/wait-signal.rs
+++ b/examples/wait-signal.rs
@@ -20,7 +20,7 @@ mod example {
         let poller = Poller::new().unwrap();
 
         // Register SIGINT in the poller.
-        let sigint = Signal(libc::SIGINT);
+        let sigint = Signal::Int;
         poller.add_filter(sigint, 1, PollMode::Oneshot).unwrap();
 
         let mut events = Events::new();
diff --git a/src/os/kqueue.rs b/src/os/kqueue.rs
index 982640b..78af91e 100644
--- a/src/os/kqueue.rs
+++ b/src/os/kqueue.rs
@@ -37,7 +37,7 @@ pub trait PollerKqueueExt<F: Filter>: PollerSealed {
     /// let poller = Poller::new().unwrap();
     ///
     /// // Register the SIGINT signal.
-    /// poller.add_filter(Signal(libc::SIGINT), 0, PollMode::Oneshot).unwrap();
+    /// poller.add_filter(Signal::Int, 0, PollMode::Oneshot).unwrap();
     ///
     /// // Wait for the signal.
     /// let mut events = Events::new();
@@ -60,10 +60,10 @@ pub trait PollerKqueueExt<F: Filter>: PollerSealed {
     /// let poller = Poller::new().unwrap();
     ///
     /// // Register the SIGINT signal.
-    /// poller.add_filter(Signal(libc::SIGINT), 0, PollMode::Oneshot).unwrap();
+    /// poller.add_filter(Signal::Int, 0, PollMode::Oneshot).unwrap();
     ///
     /// // Re-register with a different key.
-    /// poller.modify_filter(Signal(libc::SIGINT), 1, PollMode::Oneshot).unwrap();
+    /// poller.modify_filter(Signal::Int, 1, PollMode::Oneshot).unwrap();
     ///
     /// // Wait for the signal.
     /// let mut events = Events::new();
@@ -86,10 +86,10 @@ pub trait PollerKqueueExt<F: Filter>: PollerSealed {
     /// let poller = Poller::new().unwrap();
     ///
     /// // Register the SIGINT signal.
-    /// poller.add_filter(Signal(libc::SIGINT), 0, PollMode::Oneshot).unwrap();
+    /// poller.add_filter(Signal::Int, 0, PollMode::Oneshot).unwrap();
     ///
     /// // Remove the filter.
-    /// poller.delete_filter(Signal(libc::SIGINT)).unwrap();
+    /// poller.delete_filter(Signal::Int).unwrap();
     /// ```
     fn delete_filter(&self, filter: F) -> io::Result<()>;
 }
@@ -134,15 +134,14 @@ impl<T: Filter + ?Sized> Filter for &T {}
 ///
 /// No matter what `PollMode` is specified, this filter will always be
 /// oneshot-only.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
-pub struct Signal(pub std::os::raw::c_int);
+pub type Signal = rustix::process::Signal;
 
 unsafe impl FilterSealed for Signal {
     #[inline(always)]
     fn filter(&self, flags: kqueue::EventFlags, key: usize) -> kqueue::Event {
         kqueue::Event::new(
             kqueue::EventFilter::Signal {
-                signal: rustix::process::Signal::from_raw(self.0).expect("invalid signal number"),
+                signal: self.0,
                 times: 0,
             },
             flags | kqueue::EventFlags::RECEIPT,
```

**Comment by fogti** [comment]
(the doctests would also need to be fixed...)

**Comment by fogti** [comment]
https://github.com/smol-rs/polling/tree/fogti/no-libc

**Comment by notgull** [comment]
`rustix` is an unstable crate with pretty frequent breaking changes. We can't export anything from that crate.

File: Cargo.toml [hunk 1]
```diff
@@ -49,5 +49,3 @@ features = [
 easy-parallel = "3.1.0"
 fastrand = "2.0.0"
```
Comment (fogti) [inline Cargo.toml:51]: stray empty line

**Review by fogti — APPROVED** [review]

**Pull request merged**